fleetspeak/src/server/components/components.go

```diff
@@ -97,12 +98,26 @@ func MakeComponents(cfg *cpb.Config) (*server.Components, error) {
 			Listener:         l,
 			Cert:             []byte(hcfg.Certificates),
 			ClientCertHeader: hcfg.ClientCertificateHeader,
+			FrontendMode:     hcfg.FrontendMode,
 			Key:              []byte(hcfg.Key),
 			Streaming:        !hcfg.DisableStreaming,
 		})
 		if err != nil {
 			return nil, fmt.Errorf("failed to create communicator: %v", err)
 		}
+		if hcfg.FrontendMode != cpb.FrontendMode_MTLS {
+			log.Warningln("####################################################################")
+			log.Warningln("# Note:                                                            #")
+			log.Warningln("#  Your are running Fleetspeak in a frontend mode other than mTLS. #")
+			log.Warningln("#  This only makes sense if you run Fleetspeak frontends behind a  #")
+			log.Warningln("#  TLS-terminating load balancer.                                  #")
+			log.Warningln("####################################################################")
+		}
+		if (hcfg.FrontendMode == cpb.FrontendMode_MTLS && hcfg.ClientCertificateHeader != "") ||
+		   (hcfg.FrontendMode == cpb.FrontendMode_HEADER_TLS && hcfg.ClientCertificateHeader =="") {
```

Comment:
Formatting nit: =="" -> == "" (note the space between equality operator and the quotes).